protocol/0088-PPRF-party_profile.md

```diff
@@ -0,0 +1,44 @@
+# On-chain Party Profile
+
+The on-chain party profile feature allows users to associate a unique alias and free-form metadata with a public key they own. This data is stored on-chain and is exposed to anyone through APIs.
+
+
+## Updating a party profile
+
+Any user with a public key is able to update their party profile by submitting a signed `UpdatePartyProfile` transaction.
+
+To prevent spam attacks, the number of transactions which can be sent in an epoch per party is restricted and a parties total balance across all assets (expressed in quantum) must exceed `spam.protection.updatePartyProfile.min.funds`. Refer to the [spam protection specification](./0062-SPAM-spam_protection.md#party-profile-spam) for further details on these restrictions.
+
+An `UpdatePartyProfile` has the following fields.
+
+- `alias`: an optional string which can be as an alternative identifier for the public key in any dApp (e.g. in competition leaderboards).
+- `metadata`: an optional list of metadata key value pairs to further describe a public key.
+
+If metadata is specified, the metadata must adhere to the following rules:
+
+- no more than 10 key value pairs can be specified
+- a key must be no longer than 33 characters
+- a value must be no longer than 256 characters
```

Comment:
What is the reasonning for these requirements?

Reply:
Matching what was spoken about in the [slack thread](https://vegaprotocol.slack.com/archives/C02KVKMAE82/p1704729335940369?thread_ts=1704711479.332909&cid=C02KVKMAE82) and implemented in the [PR](https://github.com/vegaprotocol/vega/pull/10340/files#diff-b0254a8b84ccb6f3b38b2d955a45acbd13f1bacb208fc51a38cddbf9d805f549) 

Believe this was a memory requirement and to stop users spamming the database with enormous metadata.